### PR TITLE
Improve webpack config

### PR DESCRIPTION
**Pull request by Shelob9**
#3239 Had these goals:

* Remove vue support - once #3212 is done. This was not done, though Vue is no longer being compiled.
* Upgrade babel and webpack DONE!
* Use @wordpress/scripts for the blocks compilation DONE!
- [x] Exclude React from our front-end builds and block-builds. Include React for our non block admin views. 
 - HMR DONE!
 - Lazyloading POSSIBLE

Documentation on how JavaScript development now works can be found here:
https://github.com/CalderaWP/Caldera-Forms/blob/340ff45789030ef27a51fcb1a8835a21b4892e76/contributing/local-dev.md#javascript-development

### CODE REVIEW

**Comment by Shelob9** [comment]
Should be redone based on this: https://github.com/Shelob9/testing-react-wordpress

**Comment by Shelob9** [comment]
Ok, I just merged develop and tested this.

JavaScript tests pass.

I did a rebuild for production `yarn build`

I'm testing, and seeing no console errors or bad function when creating a contact form, editing it and making it send. The privacy settings screen (also a front-end client powered by React) works. The Pro setting screens and tabs, which BTW can no longer be compiled beacuse Vue is out of the tool chain, still work. Still works if I add a file field.

This looks good except Render client is being built without removing WordPress dependencies, which is bad.

**Comment by Shelob9** [comment]
I'm happy with this now. The render client is built without WordPress dependencies, and then we use the JSON file of dependencies output by that process as the dependencies in wp_enqueue_script. It's neato that this stuff works so well now.

@kstover Once this is merged, I'm comfortable using the new `@calderajs/*` packages in the Caldera Forms admin.

**Comment by Shelob9** [comment]
@New0 @kjohnson Can we merge this?

**Comment by Shelob9** [comment]
@New0 Getting this PR merged is going to be neccsassary before we start work on the next build cycle. Can you please take a look at this as soon as possible?

**Comment by Shelob9** [comment]
I just pulled this down and did a quick test, before pulling latest develop in. 

Looks good in form editor, form admin, entry viewer and form front-end except:

- `Warning: file_get_contents(/var/www/html/wp-content/plugins/caldera-forms/clients/render/build/index.min.deps.json): failed to open stream: No such file or directory in /var/www/html/wp-content/plugins/caldera-forms/classes/render/assets.php on line 716` in front-end above form.
- When editing an entry, the form is not loaded properly.

**Comment by Shelob9** [comment]
Commit e869090 merges in develop branch. Still works accept the same warning as before.

Also, basically all tests fail beacuse of error `TypeError: Cannot read property 'current' of undefined` thrown by reaact-test-renderer.

**Comment by Shelob9** [comment]
I lost the last comment from when I looked at this. The lingering problem was correct way to include React. Tests and front-end fail when using file fields as React is not being loaded yet.

When I wrote this in May, we needed React hooks in our admin page, and WordPress React didn't have hooks, which are React 16.8 plus

I'm looking at WordPress' package.json, I see:
* master/  5.4 -"@wordpress/element": "2.8.2"
* 5.3 - "react": "16.9.0",
* 5.2 - "react": "16.8.4",
* 5.1 - "react": "16.6.3",

[This](https://make.wordpress.org/core/2019/12/19/dev-chat-summary-december-18-2019/) gives an ETA of March 31 for WordPress 5.4

I don't want to ship my own React like this PR currently does. 

@New0 Can we use the leagacy bundle for one more release to fix this? That's a solution. Another one is setting the tested up to meta tag to "5.2".

**Comment by Shelob9** [comment]
@New0 This now works in privacy, amdin, editor, entries, front-end and tests pass. I feel accomplished and shiny.

I removed a lot of complexity. We ONLY use WordPress' React now. What's cool is that WordPress generates a list of dependencies as JSON, which we use here: https://github.com/CalderaWP/Caldera-Forms/pull/3240/files#diff-7f01e925875e64c2a1b90354ab910155R707-R716

BTW I was getting an error from Dropzone, AFTER I fixed the React issue.  The problem was we got the latest version of Dropzone installed. I forced it to use 6.2.4. Latest version uses React hooks, which is cool, but broke backwards compat.

https://github.com/CalderaWP/Caldera-Forms/pull/3240/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L58-R62

**Comment by Shelob9** [comment]
@New0 I am assigning to you know. I **think** waht needs done here is making sure legacy bundle provides backwards compat for WordPress 5.0-5.1.

Once that's done, remove the WIP prefix I just added and ask @kjohnson to take a look now that we've both touched this.

File: clients/tests/unit/render/Fields/Input.test.js [hunk 1]
```diff
@@ -11,7 +12,8 @@ describe( 'Input field', () => {
 		fieldPlaceHolder: "",
 		fieldValue: "",
 		outterIdAttr: "cf2-fld_5843941_1",
-		required: false,
+		required: false
+		,
```
Comment (kjohnson) [inline clients/tests/unit/render/Fields/Input.test.js]: Comma on separate line.

Comment (Shelob9) [reply clients/tests/unit/render/Fields/Input.test.js]: Thanks Kyle.

**Comment by Shelob9** [comment]
PR to add JS and PHP unit test automation back: https://github.com/CalderaWP/Caldera-Forms/pull/3438#issuecomment-572243920 Requires this, so build works properly.

**Pull request merged**